docs/use_cases/retrieval_augmented_generation_eval.md

```diff
@@ -0,0 +1,181 @@
+
+We could either leverage the results here (provided our model belongs to this list) or run relevant 'tasks' for our custom model (instructions provided in the github [link](https://github.com/embeddings-benchmark/mteb#leaderboard))
+
+```python
+import logging
+
+from mteb import MTEB
+from sentence_transformers import SentenceTransformer
+
+
+logging.basicConfig(level=logging.INFO)
+
+
+model_name = "average_word_embeddings_komninos"
+model = SentenceTransformer(model_name)
+evaluation = MTEB(task_langs=["en"])
+evaluation.run(model, output_folder=f"results/{model_name}", eval_splits=["test"])
+
+print("--DONE--")
+```
+
+### Data Ingestion Evaluation
+
+We first validate the model, and train it on the language of our domain. 
+We can then configure data ingestion into our semantic retrieval store aka vector store.
+Various vector databases offer index configurations to influence and enhance the retrieval quality, based on the supported index types (Flat , LSH , HNSW and IVF).
+One such example here to improve HNSW [retrieval-quality](https://qdrant.tech/documentation/tutorials/retrieval-quality/).
+
+To evaluate ingestion, we need to focus on related variables, such as:
+
+* **Chunk size** - represents the size of each segment. This depends on the token limit of our embedding model. It also has a substantial impact on the contextual understanding of the data. That impacts the precision, recall, and relevancy of our results.
+
+* **Chunk overlap** - represents the presence of overlapping information fragments in each segment. This helps with context retention of the information chunks but at the same time must be used with relevant strategies like deduplication, and content normalization to eradicate adverse effects.
+
+* **Chunking/Text splitting strategy** - represents the process of data splitting and further treatment as mentioned above.
+
+A [utility](https://chunkviz.up.railway.app/) like this seems useful to visualise your apparent chunks.
+
+### Semantic Retrieval Evaluation
+
+The next step is semantic retrieval evaluation. It puts the work you've done so far to a litmus test.
+
+Retrieval is a vital component of RAG and its evaluation can be addressed as a classic information retrieval evaluation problem.
+
+You need to establish the expectations from the returned results. This can help
+us to identify reference metrics and important parameters. Done correctly, it
+helps us determine if the documents retrieved at this stage are relevant
+results.
+
+We have several existing metrics to guide and define our baseline:
+
+- Precision and Recall or their combination F1 Score
+- DCG and nDCG, which relates to their relevance, based on the inclusion or rank of documents in the results
+
+The nature of semantic information retrieval poses a challenge at this stage, as the documents are retrieved beyond the keywords/synonyms/token enrichment-matching.
```

Comment:
I don't undestand this sentence, I guess readers won't either

Reply:
```suggestion
At this stage, the challenge arises in semantic information retrieval, where documents are retrieved based on criteria extending beyond traditional keyword, synonym, or token-level enrichment matching
```

Reply:
Does this seem better?

Reply:
Still feels a bit convoluted sentence but now I understand that it is about the bag-of-word(token) or semantic vector based representation of text :) This is I think fine by me!